projects/uswds-components/src/lib/input/affix.directive.ts

```diff
@@ -0,0 +1,79 @@
+import { AfterViewInit, Directive, ElementRef, Input, Renderer2, TemplateRef, ViewContainerRef } from '@angular/core';
+
+@Directive({
+  selector: 'input[prefix], input[suffix]'
+})
+export class UsaInputAffixDirective implements AfterViewInit {
+
+  @Input() prefix: TemplateRef<any> | string;
+  @Input() suffix: TemplateRef<any> | string;
+
+  constructor(
+    private el: ElementRef,
+    private renderer: Renderer2,
+    private viewContainer: ViewContainerRef) {
+  }
+
+  ngAfterViewInit(){
+
+    const parent = this.renderer.parentNode(this.el.nativeElement);
+
+    //Need next sibling to ensure generated usa-input-group is inserted into the same place on the page and input is expected to be
+    const nextSibling = this.renderer.nextSibling(this.el.nativeElement);
+
+    let prefixElement, suffixElement;
+    if(this.prefix){
+      prefixElement = this.handleAffix(this.prefix, 'usa-input-prefix')
+    }
+    if(this.suffix){
+      suffixElement = this.handleAffix(this.suffix, 'usa-input-suffix')
+    }
```

Comment:
What happens if neither prefix nor suffix is provided? Can we just return early at the start of this function?

Reply:
Nothing happens visually, although the input would still be placed inside `usa-input-group`, probably not ideal. Updated it to return early.